### PR TITLE
Add HEART workbook

### PR DESCRIPTION
**Pull request by lotsahelp**
## PR Checklist

* [x] Explain your changes, so people looking at the PR know *what* and *why*, the code changes are the *how*.
Add HEART metrics workbook to enable further insights on Engagement, Adoption and Retention.

image

### If adding or updating templates:
* [x] post a screenshot of templates and/or gallery changes
* [x] ensure your template has a corresponding gallery entry in the gallery folder
* [x] If you are adding a new template, add your team and template/gallery file(s) to the CODEOWNERS file. CODEOWNERS entries should be teams, not individuals
* [x] ensure all steps have meaningful names
* [ ] ensure all parameters and grid columns have display names set so they can be localized
* [ ] ensure that parameters id values are unique __or they will fail PR validation__ (parameter ids are used for localization)
* [ ] ensure that steps names are unique __or they will fail PR validation__ (step names are used for localization)
* [x] grep `/subscription/` and ensure that your parameters don't have any hardcoded resourceIds __or they will fail PR validation__
* [x] remove `fallbackResourceIds` and `fromTemplateId` fields from your template workbook __or they will fail PR validation__

### CODE REVIEW

**Review by gardnerjr — COMMENTED** [review]
one thing i see right away:
image

time range is hidden and set to 90d for default?  for large resources (like the one i tried it on first just because i was there, the queries take several minutes to complete.

you probably don't want to hide the time range unless really necessary, and it should probably not be 90d by default.

**Comment by jpurpura** [comment]
Nice catch on the Time Range parameter - we've updated the default to "Last 2 calendar months" and made this parameter visible on all pages.

**Comment by mattmccleary** [comment]
@gardnerjr The team made some query optimizations. Any other blockers for approval?

@OsvaldoRosado What's required from your side to get App Insights product team approval? Here's a [HEART Workbook](https://ms.portal.azure.com/#blade/AppInsightsExtension/UsageNotebookBlade/ComponentId/%2Fsubscriptions%2F96eced6b-b9cc-4575-99ad-ff12cf25085a%2FresourceGroups%2FHEART-Scorecard-NonProd%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2FHeartAppInsightsDemoInstance/ConfigurationId/%2Fsubscriptions%2F96eced6b-b9cc-4575-99ad-ff12cf25085a%2Fresourcegroups%2Fheart-scorecard-nonprod%2Fproviders%2Fmicrosoft.insights%2Fworkbooks%2F58d1e86c-9b3f-471d-a542-c1b54d86a0e0/Type/workbook/WorkbookTemplateName/HEART%20Metrics%20-%20Product%20Usage%20Measurement) demo that you can walk through to get a feel for it.

**Review by casocha — APPROVED** [review]
Didn't see any errors when loading the workbook & noticed the requested updates from Friday.

**Comment by casocha** [comment]
Code approved

**Review by casocha — APPROVED** [review]
Approving PR

**Comment by hakhooa** [comment]
@OsvaldoRosado : All our changes from the feedback are incorporated. Need one approval from "microsoft/applicationinsights-portalexperiences". Requesting approval.

File: gallery/usage/microsoft.insights-components.json [hunk 1]
```diff
@@ -55,6 +55,12 @@
 					"id": "Workbooks/Usage/User Retention Analysis",
 					"author": "Microsoft",
 					"name": "User Retention Analysis"
+				},
+				{
+					"id": "Workbooks/Usage/Heart",
+					"author": "Microsoft",
+					"name": "Heart - Product Usage Measurement",
```
Comment (OsvaldoRosado) [inline gallery/usage/microsoft.insights-components.json:62]: There was feedback provided offline to add some clarity in the gallery title for users to know this is specific to the click analytics plugin (ex. `Product Usage Measurement for Click Analytics`). Do we plan to change it or was it decided this is the way it should be?

**Review by OsvaldoRosado — APPROVED** [review]

File: Workbooks/Usage/Heart/Heart.workbook [hunk 2]
```diff
@@ -0,0 +1,2464 @@
...
+            "type": 2,
+            "isRequired": true,
+            "query": "let Table=datatable(value:timespan, text:string, item_order:long)[\r\n24h, 'Last 24 hours', 1,\r\n3d, 'Last 3 days', 2,\r\n7d, 'Last 7 days', 3,\r\n30d, 'Last 30 days', 4,\r\n60d, 'Last 60 days', 5,\r\n90d, 'Last 90 days', 5,\r\n];\r\nlet View_1 = view () { print value=startofmonth(now()), text=\"This calendar month\", item_order=6};\r\nlet View_2 = view () { print value=startofmonth(datetime_add(\"month\", -1, now())), text=\"Last 2 calendar months\", item_order=7};\r\nlet View_3 = view () { print value=startofmonth(datetime_add(\"month\", -2, now())), text=\"Last 3 calendar months\", item_order=8};\r\nlet View_6 = view () { print value=startofmonth(datetime_add(\"month\", -6, now())), text=\"Last 6 calendar months\", item_order=9};\r\nlet View_13 = view () { print value=startofmonth(datetime_add(\"month\", -13, now())), text=\"Last 13 calendar months\", item_order=10};\r\nTable\r\n| project value=now() - value, text, item_order\r\n| union View_1, View_2, View_3, View_6, View_13\r\n| order by item_order asc\r\n| project value, text",
+            "value": "2021-03-01T00:00:00Z",
```
[the remark this answers is not in the document]
Comment (jpurpura) [reply Workbooks/Usage/Heart/Heart.workbook]: Hello, thanks for the call out. The default has now been changed back to "Last 2 calendar months".

**Comment by jpurpura** [comment]
Hi @hakhooa @OsvaldoRosado @yves-chan @casocha - made some improvements to the workbook, let me know if you have any feedback! We are hoping to get into Private Preview in early July. Thank you!
- edited descriptions in the following tabs (for added clarity): Summary, Adoption, Engagement, Happiness, Feature Metrics
- added "Estimating Happiness" section to Happiness tab: uses telemetry to estimate user satisfaction 
- replaced Monthly/Daily tab with a json parameter, removing duplicate charts (no more separate charts for day and month) and adding a weekly option
- general spelling / grammatical edits

**Review by cartersocha — APPROVED** [review]
No data comments & warnings are updated. Workbook had no problems when I loaded it & clicked through.

**Review by gardnerjr — APPROVED** [review]
only reviewed the CODEOWNERS changes

**Pull request merged**